### PR TITLE
Walking over syringes without protection injects

### PR DESCRIPTION
**Pull request by Cocacolagua**
## What Does This PR Do
Stepping over syringes without protection injects the chemicals in the syringe in you.
!!!doesn't apply to genetics dna injectors!!!
!!!you wont get injected if you have shoes/fly/float/are buckled/are an IPC/are wearing something that covers your feet like a hardsuit/are pierce immune!!!

## Why It's Good For The Game
Kinda funny.
Promotes managing your waste disposal well.

## Images of changes
https://user-images.githubusercontent.com/48032385/232134126-50767118-c958-40c5-9b61-cc1a60794740.mp4

## Testing
1. Waited for code to compile.
2. Stepped over a bunch of syringes.
3. Code finished compiling, tried it in-game.
4. Tested with golems, xenos, IPCs, humans, humans with shoes, humans with hardsuits, humans crawling.

## Changelog
:cl:
tweak: Stepping over syringes without protection injects you their chemicals.
/:cl:

### CODE REVIEW

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 1]
```diff
@@ -178,6 +178,21 @@
 		M.update_inv_l_hand()
 		M.update_inv_r_hand()
 
+/obj/item/reagent_containers/syringe/Crossed(AM as mob|obj, oldloc)
+	var/mob/living/carbon/human/H = AM
```
Comment (GDNgit) [inline code/modules/reagents/reagent_containers/syringes.dm:182]: Why not scope this just to /mob/living/carbon/human/AM in the proc to cut the as + reasignment and to simplify this to just an istype() check

Comment (Sirryan2002) [reply code/modules/reagents/reagent_containers/syringes.dm:182]: its best to just be consistent with the parent proc

Comment (Cocacolagua) [reply code/modules/reagents/reagent_containers/syringes.dm:182]: because i'm on my evil grind

Comment (GDNgit) [reply code/modules/reagents/reagent_containers/syringes.dm:182]: the ishuman later in this proc can just be changed to istype in that case, as it already has scoped itself to human

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 2]
```diff
@@ -178,6 +178,20 @@
 		M.update_inv_l_hand()
 		M.update_inv_r_hand()
 
+/obj/item/reagent_containers/syringe/Crossed(mob/living/carbon/human/H, oldloc)
+	if(H.reagents && !(HAS_TRAIT(H,TRAIT_PIERCEIMMUNE) || ismachineperson(H) || H.floating || H.flying || H.buckled || H.shoes || (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET))))
```
Comment (lewcc) [inline code/modules/reagents/reagent_containers/syringes.dm:182]: There are a lot of negative checks here. Could you please make this an early return that isn't negated?
Something (roughly) like:
```
if(!H.reagents)
    return
if(HAS_TRAIT(...) || ismachineperson(H) || ....)
    return
H.visible_message(....)
...
```

Would make it a lot easier to understand.

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 3]
```diff
@@ -178,6 +178,20 @@
...
+		H.visible_message("<span class='danger'>[H] is injected by \the [src].</span>", \
+					"<span class='userdanger'>You are injected by \the [src]!</span>")
```
Comment (lewcc) [inline code/modules/reagents/reagent_containers/syringes.dm:184]: `The` gets automatically added by `src` here.
```suggestion
		H.visible_message("<span class='danger'>[H] is injected by [src].</span>", \
					"<span class='userdanger'>You are injected by [src]!</span>")
```

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 4]
```diff
@@ -178,6 +178,20 @@
...
+		if(IS_HORIZONTAL(H))
+			H.apply_damage(5, BRUTE,"chest")
+		else
+			H.apply_damage(5, BRUTE,(pick("l_foot", "r_foot")))
```
Comment (lewcc) [inline code/modules/reagents/reagent_containers/syringes.dm:188]:
```suggestion
			H.apply_damage(5, BRUTE, BODY_ZONE_CHEST)
		else
			H.apply_damage(5, BRUTE, (pick(BODY_ZONE_PRECISE_L_FOOT, BODY_ZONE_PRECISE_R_FOOT)))
```

**Comment by Contrabang** [comment]
I'm wondering if this should use the `/datum/component/caltrop` component, as a lot of the code is similar.

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 5]
```diff
@@ -178,6 +178,30 @@
 		M.update_inv_l_hand()
 		M.update_inv_r_hand()
 
+/obj/item/reagent_containers/syringe/Crossed(mob/living/carbon/human/H, oldloc)
+	if(!istype(H) || !H.reagents || (HAS_TRAIT(H,TRAIT_PIERCEIMMUNE) || ismachineperson(H)))
```
Comment (hal9000PR) [inline code/modules/reagents/reagent_containers/syringes.dm:182]:
```suggestion
	if(!istype(H) || !H.reagents || HAS_TRAIT(H, TRAIT_PIERCEIMMUNE) || ismachineperson(H))
```

**Comment by meow20** [comment]
Can't wait to drop 20 syringes full of poison in 1 tile wide corridor when sec chases me

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 6]
```diff
@@ -178,6 +178,30 @@
...
+	if(H.shoes || (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET)))
+		return
+
+	H.visible_message("<span class='danger'>[H] is injected by [src].</span>", \
+				"<span class='userdanger'>You are injected by [src]!</span>")
+
+	if(IS_HORIZONTAL(H))
+		H.apply_damage(5, BRUTE, BODY_ZONE_CHEST)
+	else
+		H.apply_damage(5, BRUTE,(pick(BODY_ZONE_PRECISE_L_FOOT, BODY_ZONE_PRECISE_R_FOOT)))
```
Comment (lewcc) [inline code/modules/reagents/reagent_containers/syringes.dm:197]: One last nit
```suggestion
		H.apply_damage(5, BRUTE, pick(BODY_ZONE_PRECISE_L_FOOT, BODY_ZONE_PRECISE_R_FOOT))
```

File: code/modules/reagents/reagent_containers/syringes.dm [hunk 7]
```diff
@@ -178,6 +178,30 @@
...
 
+/obj/item/reagent_containers/syringe/Crossed(mob/living/carbon/human/H, oldloc)
+	if(!istype(H) || !H.reagents || HAS_TRAIT(H, TRAIT_PIERCEIMMUNE) || ismachineperson(H))
+		return
+
+	if(H.floating || H.flying || H.buckled)
+		return
+
+	if(H.shoes || (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET)))
+		return
```
Comment (hal9000PR) [inline code/modules/reagents/reagent_containers/syringes.dm:189]: surely shoes should only protect you if you are standing

**Comment by Fox-McCloud** [comment]
This is kinda funny, but I see a lot of abuseable pathways for this---especially for keeping someone locked down in an area.

It's definitely *funny*, but I don't think this is a good idea, personally.

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**